sequel/benchmarks/driver.rb

```diff
@@ -0,0 +1,194 @@
+
+postgres_tcp_addr = ENV['POSTGRES_PORT_5432_TCP_ADDR'] || 'localhost'
+postgres_port = ENV['POSTGRES_PORT_5432_TCP_PORT'] || 5432
+mysql_tcp_addr = ENV['MYSQL_PORT_3306_TCP_ADDR'] || 'localhost'
+mysql_port = ENV['MYSQL_PORT_3306_TCP_PORT'] || 3306
+
+DATABASE_URLS = {
+  psql: "postgres://postgres@#{postgres_tcp_addr}:#{postgres_port}/rubybench",
+  mysql: "mysql2://root@#{mysql_tcp_addr}:#{mysql_port}/rubybench",
+}
+
+class BenchmarkDriver
+  def self.benchmark(options)
+    self.new(options).run
+  end
+
+  def initialize(options)
+    @repeat_count = options[:repeat_count]
+    @pattern = options[:pattern]
+  end
+
+  def run
+    files.each do |path|
+      next if !@pattern.empty? && /#{@pattern.join('|')}/ !~ File.basename(path)
+
+      if path.match(/sequel/)
+        DATABASE_URLS.each do |database, url|
+          run_single(path, connection: url, database: database)
+        end
+      else
+        run_single(path)
+      end
+    end
+  end
+
+  private
+
+
+  def measure_sequel(script, connection, path, database)
+    connection_string = Proc.new do |prepared_statements|
+      "DATABASE_URL=#{connection}?prepared_statements=#{prepared_statements} #{script}"
+    end
+
+    with_prepared_statements = measure(connection_string.call(true))
+    return unless with_prepared_statements
+    without_prepared_statements = measure(connection_string.call(false))
+    return unless without_prepared_statements
+
+    form_data = default_form_data(with_prepared_statements, path, database)
+
+    submit_request(form_data, {
+      "benchmark_run[result][with_prepared_statements]" => with_prepared_statements["iterations_per_second"].round(3),
+      "benchmark_run[result][without_prepared_statements]" => without_prepared_statements["iterations_per_second"].round(3),
+      'benchmark_result_type[name]' => 'Number of iterations per second',
+      'benchmark_result_type[unit]' => 'Iterations per second'
+    })
+
+    submit_request(form_data, {
+      "benchmark_run[result][with_prepared_statements]" => with_prepared_statements["total_allocated_objects_per_iteration"],
+      "benchmark_run[result][without_prepared_statements]" => without_prepared_statements["total_allocated_objects_per_iteration"],
+      'benchmark_result_type[name]' => 'Allocated objects',
+      'benchmark_result_type[unit]' => 'Objects'
+    })
+  end
+
+  def generate_request
+    request = Net::HTTP::Post.new('/benchmark_runs')
+    request.basic_auth(ENV["API_NAME"], ENV["API_PASSWORD"])
+    request
+  end
+
+  def default_form_data(output, path, database)
+    data = {
+      'benchmark_type[category]' => output["label"],
+      'benchmark_type[script_url]' => "#{RAW_URL}#{Pathname.new(path).basename}",
+      'benchmark_type[digest]' => generate_digest(path, database),
+      'benchmark_run[environment]' => "#{`ruby -v`}",
+      'repo' => 'sequel',
+      'organization' => 'sequel'
+    }
+    
+    if(ENV['RAILS_COMMIT_HASH'])
```

Comment:
We are benchmarking sequel here. So Rails shouldn't be the variable. Instead this should be `SEQUEL_COMMIT_HASH`